tests/functional/api/test_groups.py

```diff
@@ -304,3 +304,12 @@ def test_group_transfer(gl, group):
 
     transferred_group = gl.groups.get(transfer_group.id)
     assert transferred_group.path == transferred_group.full_path
+
+
+@pytest.mark.gitlab_premium
+@pytest.mark.xfail(reason="need to setup an identity provider or it's mock")
+def test_group_saml_group_links(gl):
+    group = gl.groups.create({"name": "gitlab-test-group", "path": "gitlab-test-group"})
```

Comment:
Does it worth to add a check that the group was created? Maybe there is a rase condition here.

Reply:
Actually, I'd suggest reusing our `group` fixture and not creating groups inside the test case. Unless there was a reason to create a new one @abhiandthetruth?

```diff
-    def test_group_saml_group_links(gl):
-    group = gl.groups.create({"name": "gitlab-test-group", "path": "gitlab-test-group"})
+    def test_group_saml_group_links(group):
```

See for example the LDAP group links tests:
https://github.com/python-gitlab/python-gitlab/blob/863ea11e2fbea3273c9177eb04df00a37bac6044/tests/functional/api/test_groups.py#L147-L172

Reply:
ohh this. Sorry I was just testing if every method is giving 401 in this particular test lol. So tried creating a group in the functional test. Was using the fixture initially. Will fix, thanks!

Reply:
@lmilbaum @nejch fixed!